### ISSUE TITLE
Loss curve

### ISSUE THREAD

**Issue by ruizhecao96** [thuml/Debiased-Self-Training#1]
Hi thank you for the great work! I am now trying to use DST on my own dataset. When training the model, the cls_loss and worst_case_loss decrease normally, but the self training loss always increase, the accuracy seems ok but I don't understand why self training loss increase. Is this behavior expected?

**Comment by thucbx99** [comment]
I think this is the expected behavior. As shown in the following figure (training curve of DST on CIFAR-100)

- In the early training stage, the model generates more and more pseudo labels, leading to increase of self training loss.
- While in the later training stage, the learned representation gets more invariant to augmentations. Also, there is a trend of overfitting wrong pseudo labels.

<img width="1063" alt="image" src="https://user-images.githubusercontent.com/57670068/220763395-8afe4b5a-2f71-45c3-ae64-5873d4fe601c.png">

**Comment by ruizhecao96** [comment]
Thank you very much for your detailed explanation! However, I saw in the paper the worst case loss goes up in early stage and then goes down later. But here the worst case loss goes down quicly first and then gose up. And in my personal case the worst case loss always goes up slowly. Can you give me some hints?

**Comment by thucbx99** [comment]
- In our method, the feature extractor and worst-case estimation head are trained adversarially. The curve above plots the loss for worst-case estimation head (first goes down and then goes up). And the trend of the feature extractor is just the _**opposite**_, which is consistent with our paper. 
- For your personal case, adjusting _**warmup_iterations**_ and _**eta_prime**_ might help. Like other adversarial training methods, the balance of players is important for the final performance. Here we employ a warmup stage to first train the feature extractor, so that when the adversarial head joins the game, the hypothesis of main classifier is better.

**Comment by ruizhecao96** [comment]
Thank you very much! I will try it.
